packages/data-point/lib/trace/trace-graph.js

```diff
@@ -0,0 +1,138 @@
+const fs = require('fs')
+const Promise = require('bluebird')
+
+// this module is used to cover circular references
+const stringify = require('json-stringify-safe')
+
+const writeFileP = Promise.promisify(fs.writeFile)
+module.exports.writeFileP = writeFileP
+
+const NS_PER_SEC = 1e9
+
+const NS_PER_MS = 1000000
+
+/**
+ * @param {Number} nano nanoseconds
+ * @returns {Number} milliseconds
+ */
+function nanoToMillisecond (nano) {
+  return nano / NS_PER_MS
+}
+
+module.exports.nanoToMillisecond = nanoToMillisecond
+
+/**
+ * @param {ReducerSpec} reducer
+ * @returns {Object} Reducer summary
+ */
+function createReducerSummary (reducer) {
+  let functionName = ''
+  let functionBody = ''
+
+  if (reducer.type === 'ReducerFunction') {
+    functionName = reducer.body.name || 'anonymous'
+    functionBody = reducer.body.toString()
+  }
+
+  return {
+    id: reducer.id || '',
+    type: reducer.type,
+    name: functionName || reducer.name || '',
+    body: functionBody || ''
+  }
+}
+
+module.exports.createReducerSummary = createReducerSummary
+
+/**
+ * @param {TraceNode} node create node label
```

Comment:
"create node label" looks like the function description instead of the param description